api/pacman-api-asset/src/main/java/com/tmobile/pacman/api/asset/service/AssetServiceImpl.java

```diff
@@ -1285,4 +1286,21 @@ private int getCountOfNonCompliancePolicies(String assetGroup) {
             return 0;
         }
     }
+
+    public DatasourceData getAssetGroupsDataForShipperByDatasource(final String datasource) {
+        DatasourceData datasourceData = new DatasourceData();
+        Map<String, List<String>> assetGroupDomains = new HashMap<>();
+        List<String> targetTypes = repository.getTargetTypesByDatasource(datasource);
+        List<String> indices = targetTypes.stream().map(targetType ->
+                datasource + Constants.DELIMITER_UNDERSCORE + targetType).collect(Collectors.toList());
+        List<String> aliases = repository.getAliasByIndices(indices);
+        List<String> assetGroups = repository.getVisibleAssetGroupsFiltered(aliases);
+        assetGroups.forEach(assetGroup -> {
+            assetGroupDomains.put(assetGroup, getDomains(assetGroup));

```

Comment:
let's make domain a constant and get rid of OS and DB calls.
Pls add TODO with description that this hardcoded value will be removed and that we need to get rid of domain as a concept.

Reply:
added